contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -3132,11 +3153,46 @@ bbf_object_access_hook(ObjectAccessType access, Oid classId, Oid objectId, int s
 		(*prev_object_access_hook) (access, classId, objectId, subId, arg);
 
 	if (access == OAT_DROP && classId == RelationRelationId)
-		pltsql_drop_view_definition(objectId);
+	{	
+		char relkind;
+		relkind = get_rel_relkind(objectId);
 
+		/* Only call pltsql_drop_view_definition for views */
+		if (relkind == RELKIND_VIEW)
+			pltsql_drop_view_definition(objectId);
+		
+		/* Handle view dependencies for tables and views */
+		if (sql_dialect == SQL_DIALECT_TSQL && 
+			(relkind == RELKIND_RELATION || relkind == RELKIND_VIEW))
+		{
+			ObjectAddress obj;
+
+			obj.classId = RelationRelationId;
+			obj.objectId = objectId;
+			obj.objectSubId = 0;
+
+			/* Call view dependency handling function */
+			bbf_view_binding_on_object_drop(&obj, NULL, NULL);
+		}
+	}
 	if (access == OAT_DROP && classId == ProcedureRelationId)
-		pltsql_drop_func_default_positions(objectId);
+	{
+		if(subId != -1)
+			pltsql_drop_func_default_positions(objectId);
 
+		/* Handle function dependencies for views */
+		if (sql_dialect == SQL_DIALECT_TSQL && IS_TDS_CONN() && subId == -1)
+		{
+			ObjectAddress obj;
+			
+			obj.classId = ProcedureRelationId;
+			obj.objectId = objectId;
+			obj.objectSubId = 0;
+			
+			/* Call view dependency handling for functions */
+			bbf_view_binding_on_object_drop(&obj, NULL, NULL);
+		}
+	}
 	if (sql_dialect != SQL_DIALECT_TSQL)
 		return;
```

Comment:
I'd say should move all the logic after this dialect check, then you don't need to do a check inside your own logic.